In [1]:
# ejecuta caso 1
print('caso 1')

caso 1


In [1]:
# importando la api de statsmodels
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [3]:
# Importando un DataFrame de pandas.
import pandas as pd
df=pd.read_csv('DOWJONES_1.csv')
print(df[:3])

     ATT     DJIA     GM    IBM  INT1M      ratt       rdj       rgm  \
0  28.79  4793.78  36.16  23.12   5.15 -0.023406  0.006065 -0.013100   
1  28.44  4784.38  36.45  23.50   5.20 -0.012157 -0.001961  0.008020   
2  29.08  4795.94  36.55  24.22   5.16  0.022504  0.002416  0.002743   

       ribm     pratt      prdj      prgm     pribm  
0 -0.005591 -0.023549  0.006066 -0.013101 -0.005591  
1  0.016436 -0.012301 -0.001961  0.008020  0.016436  
2  0.030638  0.022360  0.002416  0.002743  0.030638  


INT1M = Tipo de interess de bonos del estado

ratt - rdj - rgm = rentabilidad diaria

pratt ,prdj, prgm, pribm = Prima de riesgo de cada uno logicamnete = (INT1M - ratt) = (INT1M /360 dias) /100 = ((INT1M /360) /100) - ratt))




In [4]:
# utilizando la api de formula de statsmodels
est = smf.ols(formula='pratt ~ prdj', data=df) # smf.ols modelo de regresion lineal simple (ols)
res = est.fit()
print(res.summary())
est = smf.ols(formula='pribm ~ prdj', data=df)
res=est.fit()
print(res.summary())
est = smf.ols(formula='prgm ~ prdj', data=df)
res=est.fit()
print(res.summary())


                            OLS Regression Results                            
Dep. Variable:                  pratt   R-squared:                       0.155
Model:                            OLS   Adj. R-squared:                  0.155
Method:                 Least Squares   F-statistic:                     239.6
Date:                Mon, 31 Oct 2022   Prob (F-statistic):           9.60e-50
Time:                        12:34:03   Log-Likelihood:                 3219.5
No. Observations:                1305   AIC:                            -6435.
Df Residuals:                    1303   BIC:                            -6425.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.0006      0.001     -0.999      0.3

- P> t = p/valor > 0.05 no Rechazo la Hipotesis
- Coef = constante
- prdj = pendiente  0.8183 defensiva <1 y viceverza
- R-squared = Bondad del ajuste = 0.155 puedo explicar un 0.155 
- bondad del ajuste (goodness of fit) explica que parte del riesgo de la accion podemos explicar con el riesgo del mercado    (downjones). Solo ese % puede explicar con el mercado, y la diferencia no se puede explicar con el mercado.

La accion no explica nada de

1. Se cumple el modelo CAPM?

2. Es una accion ofensiva o defensiva (de riesgo)?

3. Que parte del riesgo de la accion puede explicar el riesgo del marcao?